### PR TITLE
[KOGITO-9262] extend devmode probes threshold

### PR DESCRIPTION
**Pull request by ricardozanini**
**Description of the change:**
Small change to enhance devmode experience: https://issues.redhat.com/browse/KOGITO-9262

**Motivation for the change:**

**Checklist**

- [x] Add or Modify a unit test for your change
- [x] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Review by spolti — APPROVED** [review]

**Review by MarianMacik — APPROVED** [review]

File: controllers/profiles/object_creators_dev.go [hunk 1]
```diff
@@ -25,6 +26,11 @@ import (
 	operatorapi "github.com/kiegroup/kogito-serverless-operator/api/v1alpha08"
 )
 
+const (
+	// healthFailureThresholdDevMode exclusive threshold for devmode given that it might take longer than the immutable image to start/live/respond.
+	healthFailureThresholdDevMode = 50
```
Comment (radtriste) [inline controllers/profiles/object_creators_dev.go:31]: in seconds ?

**Review by radtriste — APPROVED** [review]
one minor comment

File: controllers/profiles/object_creators_dev.go (hunk 1 above)
Comment (ricardozanini) [reply controllers/profiles/object_creators_dev.go:31]: Nops, it's not a time measure, but a count. To illustrate:

```yaml
ports:
- name: liveness-port
  containerPort: 8080
  hostPort: 8080

livenessProbe:
  httpGet:
    path: /healthz
    port: liveness-port
  failureThreshold: 1
  periodSeconds: 10

startupProbe:
  httpGet:
    path: /healthz
    port: liveness-port
  failureThreshold: 30
  periodSeconds: 10
```

> Thanks to the startup probe, the application will have a maximum of 5 minutes (30 * 10 = 300s) to finish its startup. Once the startup probe has succeeded once, the liveness probe takes over to provide a fast response to container deadlocks. If the startup probe never succeeds, the container is killed after 300s and subject to the pod's restartPolicy.

Source: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-startup-probes

We should review this [here](https://issues.redhat.com/browse/KOGITO-9265) based on field feedback.

Comment (radtriste) [reply controllers/profiles/object_creators_dev.go:31]: sorry. misread ... ok then

**Pull request merged**